### ISSUE TITLE
ergoCubEmotions uses more than 100% of the cpu on ergocub-head

### ISSUE THREAD

**Issue by GiulioRomualdi** [icub-tech-iit/ergocub-software#213]
I noticed that `ergoCubEmotions` uses a really high amount of CPU on the ergocub-head.

In the following screenshot you can check the amount of used CPU (181%) only ergoCubEmotions was running on the ergocub-head
![image](https://github.com/icub-tech-iit/ergocub-software/assets/16744101/c74888f0-39c0-4280-8783-2e95a5f5e589)

I was wondering if it is normal

cc @martinaxgloria

**Comment by traversaro** [comment]
If I remember correctly once we checked that, one problem may be that at every module update, the file are re-read (see https://github.com/icub-tech-iit/ergocub-software/blob/eaacbf120b355edd931d2ae272e568b507740d2e/src/modules/ergoCubEmotions/ergoCubEmotions.cpp#L119). Perhaps we could read the images only once in the configure, and save them as cv::Mat attributes that can be used in the updateModule?

**Comment by martinaxgloria** [comment]
Hi @traversaro, 

thank you for your suggestion. Actually, we are reading videos and not images (the _static_ part is a 2 sec looped video and the transitions from one expression to another too). If I remember well, opencv shows videos one frame at the time after they are opened. I don't know if it's possible to save cv::VideoCapture elements as per cv::Mat ones. 

Today I tried to use the configuration `image -> transition -> video` and I saw from jtop that the usage is reduced to ~30% but there's a spike when the transition is called. 

Moreover, the % showed by jtop should correspond to the sum of the cpu usage for each core: in fact running `top` with IRIX mode set to OFF, this is the cpu consumption divided by the number of cores:

![image](https://github.com/icub-tech-iit/ergocub-software/assets/114698424/ee541f06-5123-43e6-9af8-8c7a705c9fca)

This means that the cores should not be saturated, I don't know if you see some delays.
I'm going to understand if it's possible to prevent the videos from being reopened at each update 

cc @Nicogene @GiulioRomualdi
